### PR TITLE
DRILL-7729: Use java.time in column accessors

### PR DESCRIPTION
**Pull request by paul-rogers**
# [DRILL-7729](https://issues.apache.org/jira/browse/DRILL-7729): Use java.time in column accessors
## Description

Uses` java.time` classes in the column accessors. Leaves Joda time for `Interval`, as it has no `java.time` equivalent.

This change allows the column accessors to work with Drill's JSON writer implementation. This PR includes a new `JsonWriter` based on a `RowSetReader`.

## Documentation

The change is transparent to users **except** in one particular case: a use of the "provided schema" feature in which the user has provided a format for a `DATE`, `TIME` or `TIMESTAMP` column. The `java.time` formats are slightly different than their Joda predecessors.

## Testing

Modified all tests which used Joda formats. Reran the full unit test suite.

### CODE REVIEW

File: exec/vector/src/main/java/org/apache/drill/exec/vector/DateUtilities.java [hunk 1]
```diff
@@ -195,7 +197,27 @@ public static int timeToMillis(int hours, int minutes, int seconds, int millis)
    * @param localTime Java local time
    * @return Drill form of the time
```
Comment (cgivre) [inline exec/vector/src/main/java/org/apache/drill/exec/vector/DateUtilities.java:198]: @paul-rogers 
Would you mind please adding a brief javadoc to this class explaining what a Drill form of the time is?

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/scan/convert/StandardConversions.java [hunk 2]
```diff
@@ -166,6 +171,12 @@ public static DirectConverter newInstance(
     try {
       final Constructor<? extends DirectConverter> ctor = conversionClass.getDeclaredConstructor(ScalarWriter.class, Map.class);
       return ctor.newInstance(baseWriter, properties);
+    } catch (final InvocationTargetException e) {
+      throw UserException.validationError(e.getTargetException())
+        .addContext("Converter setup failed")
+        .addContext("Conversion class" + conversionClass.getName())
+        // .addContext(errorContext) // Add after merge
```
Comment (cgivre) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/scan/convert/StandardConversions.java:178]: Should we add this line back in?

**Comment by vvysotskyi** [comment]
@paul-rogers, did you have a chance to take a look at @cgivre comments?

**Comment by luocooong** [comment]
@paul-rogers Hello, Is this PR still necessary to merge? Is it replaced by [#2149](https://github.com/apache/drill/pull/2149)?

**Comment by paul-rogers** [comment]
@luocooong, thanks for the reminder. Looks like I originally factored out the Java time stuff for its own review. Got no takers, it seems, and I forgot about it. So, when @cgivre asked me to finish the streaming JSON stuff, I just included the time stuff as well.

If the team prefers, I can pull the Java time stuff out of the streaming JSON PR and replace the code here with the updated version. Or, if the team is OK with the single big PR, we can keep it in the streaming JSON PR.

How would you like to proceed?

**Comment by vvysotskyi** [comment]
I propose to leave changes in bigger PR, to avoid additional work on splitting it and rebasing this one onto the master.

**Comment by paul-rogers** [comment]
@vvysotskyi, thanks for the suggestion. Given that, I'll go ahead and close this PR.

**Pull request closed without merging**